src/components/pages/schedule-page/schedule-event-edit-section/schedule-info-card/components/TextWithLinks.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FC } from 'react';
+import { Box } from '@mui/material';
+
+import Link from '@/components/common/ui/custom-link/CustomLink';
+import theme from '@/styles/theme';
+
+interface EventDisciplineLinkProps {
+  infoText?: string;
+  defaultText: string;
+}
+
+const TextWithLinks: FC<EventDisciplineLinkProps> = ({
+  infoText,
+  defaultText,
+}) => {
+  const linkifyText = (inputText: string) => {
+    const linkRegex = /(\bhttps?:\/\/\S+\b)/gi;
+    const parts = inputText.split(linkRegex);
+
+    return parts.map((part, index) => {
+      if (part.match(linkRegex)) {
+        return <Link key={index} href={part} text={part} />;
+      } else {
+        return part;
+      }
+    });
+  };
+
+  return (
+    <Box sx={{ typography: theme.typography.body1 }}>
+      {infoText ? linkifyText(infoText) : defaultText}
+    </Box>
```

Comment:
There is a problem with putting it ll in `Box`
![image](https://github.com/fictadvisor/fictadvisor-web/assets/86970828/d838792d-a957-4f78-86d4-345fd1eb4360)
As u may see, text without line character is displayed in a single line, u may consider replacing `Box` with 
`Typography` as [it](https://github.com/fictadvisor/fictadvisor-web/blob/db8e006a4b4a21a5deaded8b83951d43568b42fd/src/components/pages/schedule-page/schedule-event-edit-section/schedule-info-card/ScheduleInfoCard.tsx#L166) used to be, or invent some work around